chapters/operatorsandexpressions.tex

```diff
@@ -22,17 +22,29 @@ \section{Expressions}\label{expressions}
 
 \section{Operator Precedence and Associativity}\label{operator-precedence-and-associativity}
 
-Operator precedence determines the order of evaluation of operators in an expression.
-An operator with higher precedence is evaluated before an operator with lower precedence in the same expression.
-For example, `\lstinline!*!' having higher precedence than `\lstinline!+!' means that \lstinline!1 + 2 * 3! becomes evaluated as \lstinline!1 + (2 * 3)!.
-
-Operator associativity are used to determine the order of evaluation when operators have equal precedence.
-Left associativity means that evaluation proceeds from left to right.
-For example, left associativity of binary `\lstinline!-!' means that \lstinline!1 - 2 - 3! becomes evaluated as \lstinline!(1 - 2) - 3!.
-Right associativity means evaluation from right to left, but there are no operators with this associativity in Modelica.
-An operator may also be non-associative, meaning that evaluation order cannot be determined based on associativity.
-For example, non-associativity of `\lstinline!<!' means that \lstinline!1 < 2 < 3! is an invalid expression.
-Note that the operators don't need to be identical for associativity to matter; also \lstinline!1 == 2 < 3! is invalid, and \lstinline!1 - 2 + 3! becomes evaluated as \lstinline!(1 - 2) + 3!.
+Operator precedence determines the implicit subexpression structure of expressions with operators.
+(Explicit subexpression structure can be expressed by wrapping the subexpression in parentheses.)
+An operator with higher precedence ties harder to its operands than an operator with lower precedence.
+For example, `\lstinline!*!' having higher precedence than `\lstinline!+!' means that \lstinline!1 + 2 * 3! is implicitly structured as \lstinline!1 + (2 * 3)!.
+
+Precedence group associativity is used to determine the implicit subexpression structure when operators belong to the same group of equal precedence.
+Left associativity means that subexpressions are formed from left to right.
+For example, left associativity of binary additive operators means that \lstinline!1 - 2 - 3! is implicitly structured as \lstinline!(1 - 2) - 3!.
+Right associativity means forming subexpressions from right to left, but there is no precedence group with this associativity in Modelica.
```

Comment:
```suggestion
```
Since we don't have any right associative operators I would propose to not define it.